vtex/faststore
# fix: Intelligent Search term redirects when available

## What's the purpose of this pull request?

This PR intends to fix the behavior when there is an Intelligent Search term redirect registered on VTEX Admin, redirecting the user on the client-side.

FastStore Experience team discussed about solving this on the server-side, but we are not going to handle the drawbacks of generate the search page (`pages/s.tsx`) as SSR using `getServerSideProps` right now. By performing this query (and the other function calls) as SSR page, each time user visits that page the potential extra costs needs to be validated later.

## How it works?

When performing a search, if the term has a redirect registered on VTEX Admin the value of the `redirect` field should be an URL, used to redirect the user to an internal or external URL.

Check https://github.com/vtex/faststore/pull/1749 for more information.

## How to test it?

- Run the `@faststore/core` local server (`yarn dev`);
- Type and run the term `tech` in the search input field, you should be redirected to the `/technology` slug page.

This happens because there is already a redirect registered for the `storeframework` account.

### Starters Deploy Preview
